AndroidSDK/src/com/leanplum/LeanplumFcmProvider.java

```diff
@@ -48,34 +45,48 @@ public boolean isInitialized() {
     return true;
   }
 
-  public boolean isManifestSetUp() {
-    boolean hasReceivers =
-        LeanplumManifestHelper.checkComponent(LeanplumManifestHelper.getReceivers(),
-            PUSH_RECEIVER, false, null,
-            Collections.singletonList(PUSH_FIREBASE_MESSAGING_SERVICE), null);
+  @Override
+  public boolean isManifestSetup() {
+    Context context = Leanplum.getContext();
+    if (context == null) {
+      return false;
+    }
 
-    boolean hasPushFirebaseMessagingService = LeanplumManifestHelper.checkComponent(
-        LeanplumManifestHelper.getServices(), PUSH_FIREBASE_MESSAGING_SERVICE, false, null,
-        Collections.singletonList(MESSAGING_EVENT), null);
-    boolean hasPushFcmListenerService = LeanplumManifestHelper.checkComponent(
-        LeanplumManifestHelper.getServices(), PUSH_FCM_LISTENER_SERVICE, false, null,
-        Collections.singletonList(INSTANCE_ID_EVENT), null);
-    boolean hasPushRegistrationService = LeanplumManifestHelper.checkComponent(
-        LeanplumManifestHelper.getServices(), PUSH_REGISTRATION_SERVICE, false, null, null, null);
+    if (!BuildConfig.DEBUG) {
+      return true;
+    }
+    // Firebase can only be setup through gradle, so we don't have to check manually
+    // whether manifest is properly setup. We will only check our own services.
+    try {
+      boolean hasPushFirebaseMessagingService = LeanplumManifestHelper.checkComponent(
+          LeanplumManifestHelper.ApplicationComponent.SERVICE,
+          LeanplumPushFirebaseMessagingService.class.getName(), false, null,
+          Collections.singletonList(LeanplumManifestHelper.MESSAGING_EVENT), null);
```

Comment:
Could you please put packageName as the last parameter? and please return false at checkComponent if packageName equals null.